lib/workers.js

```diff
@@ -262,15 +279,20 @@ class Workers extends EventEmitter {
   }
 
   run() {
-    return new Promise((res, rej) => {
+    recorder.startUnlessRunning();
+    recorder.add('bootstrapAll hook', () => new Promise((res, rej) => {
       runHook(this.codecept.config.bootstrapAll, () => {
-        for (const worker of this.workers) {
-          const workerThread = createWorker(worker);
-          this._listenWorkerEvents(workerThread);
-        }
         res(true);
       }, 'bootstrapAll');
+    }));
+    event.dispatcher.emit(event.workers.before);
```

Comment:
this way we can attach async hooks before workers were initialized

Reply:
@PeterNgTr this way we will be able to integrate ReportPortal with Codecept for parallel runs in worker.
See my PR https://github.com/reportportal/agent-js-codecept/pull/3

Reply:
that's awesome 👏 @DavertMik